src/GeositeFramework/Views/Home/Index.cshtml

```diff
@@ -554,25 +554,44 @@
         <div class="measure-tooltip">
             <p><%= instructions %></p>
             <div>
-                <div class="measure-label"><%= segmentLabel %>:</div><%= segmentLength %> <%= units%>
+                <div class="measure-label"><%= segmentLabel %>:</div>
+                <% _.each(segmentLengths, function (item, index) { %>
+                    <div><%= item.length %> <%= item.uom %></div>
+                <% }); %>
             </div>
             <div>
-                <div class="measure-label"><%= lengthLabel %>:</div><%= totalLength %> <%= units%>
+                <div class="measure-label"><%= lengthLabel %>:</div>
+                <% _.each(totalLengths, function (item, index) { %>
+                    <div><%= item.length %> <%= item.uom %></div>
+                <% }); %>
             </div>
         </div>
     </script>
 
     <script type="text/template" id="template-measure-infobubble">
         <div class="measure-bubble">
-            <% if (area) { %>
+            <% if (areas) { %>
                 <p>
                     <span class="measure-label i18n" data-i18n="Area:">Area:</span>
-                    <span><%= area %> <%= areaUnits%><sup>2</sup></span>
+                    <div>
+                        <% _.each(areas, function (item, index) { %>
```

Comment:
The results look correct, but they're hard to read in that popup and the delete button is obscured.  Perhaps we should display them stacked vertically and just drop the "length" for area, which I assume means perimeter (if they want to measure lenght, let them draw a line).  Then we could move the delete button up to the right side, or at least make the popup taller so that it's visible.